src/Symfony/Component/Console/Style/OutputBlock.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Style;
+
+use Symfony\Component\Console\Formatter\OutputFormatter;
+use Symfony\Component\Console\Formatter\OutputFormatterInterface;
+use Symfony\Component\Console\Helper\Helper;
+
+/**
+ * Generates formatted block lines.
+ *
+ * @author Vadim Zharkov <hushker@gmail.com>
+ */
+class OutputBlock
+{
+    /**
+     * @var OutputFormatterInterface
+     */
+    protected $formatter;
+
+    /**
+     * @var int
+     */
+    protected $lineLength;
```

Comment:
properties must be private according to our policy. We don't want to make them inheritance-based extension points (we don't do that without a clear use case and a decision from the core team that inheritance is the intended way to extend, as that's the most painful kind of extension point in regards of building compatibility layers)

Reply:
Got it. Thanks for the clear and comprehensive explanation.